### PR TITLE
[CAP-206] Refactor code routing, service

### PR DESCRIPTION
**Pull request by banhsbao**

### CODE REVIEW

File: jobfair-frontend/src/constants/Paths/Path.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+export const PATH = {
+  JOB_FAIR_PARK_PAGE: '/map/:jobFairId',
+  ATTENDANT_JOB_FAIR_PAGE: '/jobfair/attendant/:companyBoothId',
+  JOB_FAIR_REGISTRATION_PAGE_ID: '/company-register-jobfair/:jobfairId',
```
Comment (CaoSon1909) [inline jobfair-frontend/src/constants/Paths/Path.js:4]: cant use this path because of "/:jobFairId" at the end.

**Review by CaoSon1909 — COMMENTED** [review]
wow so many file changes. I'll be back to review later

**Review by CaoSon1909 — DISMISSED** [review]

File: jobfair-frontend/src/constants/Paths/Path.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}
+export const PATH_ATTENDANT = {
+  ATTENDANT_PROFILE_PAGE: '/attendant/profile',
+  APPLIED_JOB_PAGE: '/applied-job'
+}
+export const PATH_COMPANY_EMPLOYEE = {
+  COMPANY_PROFILE_PAGE: '/employee/company-profile'
+}
+export const PATH_COMPANY_MANAGER = {
+  EMPLOYEE_MANAGENT_PAGE: '/company/employee-management',
```
Comment (TienTruong2000) [inline jobfair-frontend/src/constants/Paths/Path.js:30]: Spelling error

File: jobfair-frontend/src/services/layout-controller/LayoutControllerService.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { CallAPI } from '../axiosBase'
+import { ENDPOINT_GET_LAYOUT_AND_AVAILABLE_SLOT_BY_JOB_FAIR_ID } from '../../constants/Endpoints/layout-controller/LayoutControllerEndPoint'
```
Comment (TienTruong2000) [inline jobfair-frontend/src/services/layout-controller/LayoutControllerService.js:2]: This variable does not exist

File: jobfair-frontend/src/router/StaffRouting.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import StaffRouter from './components/StaffRouter'
+import { PATH_STAFF } from '../constants/Paths/Path'
+const StaffRouting = () => {
+  return (
+    <>
+      <StaffRouter
+        key=""
+        component={() => <ApprovalRegistrationPage />}
```
Comment (TienTruong2000) [inline jobfair-frontend/src/router/StaffRouting.js:8]: Component not imported

File: jobfair-frontend/src/router/PublicRouting.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      </Route>
+      <Route path={PATH.LOGIN_PAGE} exact>
+        {/*if user has already login, avoid them to login again*/}
+        {!role ? <LoginPage /> : <Redirect to="/" />}
+      </Route>
+      <Route path={PATH.REGISTER_PAGE} exact>
+        {!role ? <RegisterPage /> : <Redirect to="/" />}
+      </Route>
+      <Route path={PATH.CHANGE_PASSWORD_PAGE} exact>
+        {role ? <ChangePasswordPage /> : <Redirect to="/auth/login" />}
```
Comment (TienTruong2000) [inline jobfair-frontend/src/router/PublicRouting.js:50]: Why dont use Path constant for Redirect?

File: jobfair-frontend/src/router/CompanyManagerRouting.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import CompanyManagerRouter from './components/CompanyManagerRouter'
+import EmployeeManagementPage from '../pages/EmployeeManagementPage/EmployeeManagementPage'
+import EmployeeRegisterPage from '../pages/EmployeeRegisterPage/EmployeeRegisterPage'
+import RegisterJobFairForm from '../components/register-job-fair-form/RegisterJobFairForm'
+import { PATH_COMPANY_MANAGER } from '../constants/Paths/Path'
+const CompanyManagerRouting = () => {
+  return (
+    <>
+      <CompanyManagerRouter
+        key="/company/employee-management"
```
Comment (TienTruong2000) [inline jobfair-frontend/src/router/CompanyManagerRouting.js:11]: Why don't use Path constant for key?

File: jobfair-frontend/src/router/CompanyManagerRouting.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+      />
+      <CompanyManagerRouter
+        key="/company/register-job-fair"
+        component={() => <RegisterJobFairForm />}
+        path={PATH_COMPANY_MANAGER.REGISTER_JOB_FAIR_FORM}
+        exact
+      />
+      <CompanyManagerRouter
+        key="/company/profile"
+        component={() => <CompanyProfile />}
```
Comment (TienTruong2000) [inline jobfair-frontend/src/router/CompanyManagerRouting.js:30]: Component not yet imported

**Review by CaoSon1909 — CHANGES_REQUESTED** [review]
image

File: jobfair-frontend/src/router/components/CompanyManagerRouter.js [hunk 8]
```diff
@@ -1,10 +1,9 @@
...
+import { COMPANY_MANAGER } from '../../constants/RoleType'
 import React from 'react'
 import { useSelector } from 'react-redux'
 import { Route, Redirect } from 'react-router-dom'
-import { contains } from '../utils/common'
-import ErrorPage from '../pages/ErrorPage/ErrorPage'
+import ErrorPage from '../../pages/ErrorPage/ErrorPage'
 
 const CompanyManagerRouter = ({ component: Component, ...rest }) => {
   const { user, isAuthUser } = useSelector(state => state.authentication)
```
Comment (TienTruong2000) [inline jobfair-frontend/src/router/components/CompanyManagerRouter.js:9]: at line 22 of this file the Redirect not yet use Path constant

File: jobfair-frontend/src/router/components/CompanyEmployeeRouter.js [hunk 9]
```diff
@@ -1,10 +1,9 @@
 //This routing rule allows the current user to have this role "COMPANY_EMPLOYEE"
-import { COMPANY_EMPLOYEE } from '../constants/RoleType'
+import { COMPANY_EMPLOYEE } from '../../constants/RoleType'
 import React from 'react'
 import { useSelector } from 'react-redux'
 import { Route, Redirect } from 'react-router-dom'
-import { contains } from '../utils/common'
-import ErrorPage from '../pages/ErrorPage/ErrorPage'
+import ErrorPage from '../../pages/ErrorPage/ErrorPage'
```
Comment (TienTruong2000) [inline jobfair-frontend/src/router/components/CompanyEmployeeRouter.js:7]: at line 23 of this file the Redirect not yet use Path constant

File: jobfair-frontend/src/router/components/AttendantRouter.js [hunk 10]
```diff
@@ -1,10 +1,9 @@
 //This routing rule allows the current user to have this role "ATTENDANT"
-import { ATTENDANT } from '../constants/RoleType'
+import { ATTENDANT } from '../../constants/RoleType'
 import React from 'react'
 import { useSelector } from 'react-redux'
 import { Route, Redirect } from 'react-router-dom'
-import { contains } from '../utils/common'
-import ErrorPage from '../pages/ErrorPage/ErrorPage'
+import ErrorPage from '../../pages/ErrorPage/ErrorPage'
```
Comment (TienTruong2000) [inline jobfair-frontend/src/router/components/AttendantRouter.js:7]: at line 22 of this file the Redirect not yet use Path constant

File: jobfair-frontend/src/router/components/AdminRouter.js [hunk 11]
```diff
@@ -1,10 +1,9 @@
 //This routing rule allows the current user to have this role "ADMIN"
-import { ADMIN } from '../constants/RoleType'
+import { ADMIN } from '../../constants/RoleType'
 import React from 'react'
 import { useSelector } from 'react-redux'
 import { Route, Redirect } from 'react-router-dom'
-import { contains } from '../utils/common'
-import ErrorPage from '../pages/ErrorPage/ErrorPage'
+import ErrorPage from '../../pages/ErrorPage/ErrorPage'
```
Comment (TienTruong2000) [inline jobfair-frontend/src/router/components/AdminRouter.js:7]: at line 23 of this file the Redirect not yet use Path constant

**Review by TienTruong2000 — CHANGES_REQUESTED** [review]

File: jobfair-frontend/src/constants/Paths/Path.js (hunk 2 above)
Comment (banhsbao) [reply jobfair-frontend/src/constants/Paths/Path.js:30]: Fixed

File: jobfair-frontend/src/services/layout-controller/LayoutControllerService.js (hunk 3 above)
Comment (banhsbao) [reply jobfair-frontend/src/services/layout-controller/LayoutControllerService.js:2]: Fixed

File: jobfair-frontend/src/router/StaffRouting.js (hunk 4 above)
Comment (banhsbao) [reply jobfair-frontend/src/router/StaffRouting.js:8]: Fixed

File: jobfair-frontend/src/router/PublicRouting.js (hunk 5 above)
Comment (banhsbao) [reply jobfair-frontend/src/router/PublicRouting.js:50]: Fixed

File: jobfair-frontend/src/router/CompanyManagerRouting.js (hunk 6 above)
Comment (banhsbao) [reply jobfair-frontend/src/router/CompanyManagerRouting.js:11]: Fixed

File: jobfair-frontend/src/router/CompanyManagerRouting.js (hunk 7 above)
Comment (banhsbao) [reply jobfair-frontend/src/router/CompanyManagerRouting.js:30]: Fixed

**Review by CaoSon1909 — APPROVED** [review]
LGTM

**Pull request merged**